zipline/gens/tradesimulation.py

```diff
@@ -266,11 +268,14 @@ def _call_handle_data(self):
         Call the user's handle_data, returning any orders placed by the algo
         during the call.
         """
+        time_start = utcnow()
         self.algo.event_manager.handle_data(
             self.algo,
             self.current_data,
             self.simulation_dt,
         )
+        time_end = utcnow()
+        self.algo.call_handle_data_duration.append((time_end-time_start).total_seconds()*1000)
```

Comment:
Also, if the seconds in between is what is desired, I'd recommend just using time.time(), avoids the overhead of creating the datetime/Timestamp/timedeltas.